File: app/src/main/java/foz/cueaside/aa/Routine.java
package foz.cueaside.aa;

import java.util.List;

public class Routine {
    public String id;
    public List<AppInfo> apps;
    public String cond; // "launched", "used", "exiting"
    public int dur;
    public String unit; // "m", "h", "s"
    public String timeMode; // "session", "total"
    public IconInfo icon;
    public String title;
    public String msg;
    public boolean bubble;
    public boolean enabled;

    public static class AppInfo {
        public String name;
        public String pkg;
        public String icon; // base64 or emoji
    }

    public static class IconInfo {
        public String type; // "app", "preset", "lib"
        public String pkg;
        public String e;
        public String src;
    }
}

File: native-port/src/main/kotlin/foz/cueaside/aa/native_port/model/Routine.kt
package foz.cueaside.aa.native_port.model

data class Routine(
    val id: String,
    val seqId: Int,
    val cueName: String,
    val apps: List<AppInfo>,
    val cond: String, // "launched", "used", "exiting"
    val dur: Int,
    val unit: String, // "m", "h", "s"
    val timeMode: String, // "session", "total"
    val icon: IconInfo? = null,
    val title: String? = null,
    val msg: String,
    val bubble: Boolean = false,
    val enabled: Boolean = true,
    val highPriority: Boolean = true,
    val timeout: Int = 0 // seconds
)

data class AppInfo(
    val name: String,
    val pkg: String,
    val icon: String // base64 or emoji
)

data class IconInfo(
    val type: String, // "app", "preset", "lib"
    val pkg: String? = null,
    val e: String? = null,
    val src: String? = null
)
